tokio/src/io/async_write.rs

```diff
@@ -127,6 +127,51 @@ pub trait AsyncWrite {
     /// This function will panic if not called within the context of a future's
     /// task.
     fn poll_shutdown(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), io::Error>>;
+
+    /// Like [`poll_write`], except that it writes from a slice of buffers.
+    ///
+    /// Data is copied from each buffer in order, with the final buffer
+    /// read from possibly being only partially consumed. This method must
+    /// behave as a call to [`write`] with the buffers concatenated would.
+    ///
+    /// The default implementation calls [`poll_write`] with either the first nonempty
+    /// buffer provided, or an empty one if none exists.
```

Comment:
should there be a note that the default implementation is less efficient than just using `poll_write` and that `poll_write_vectored` should only be called when `is_write_vectored` returns `true`? maybe including an example of checking for vectored write capability before using it?